core/formatter.cpp

```diff
@@ -1724,16 +1733,233 @@ class FixIndentation {
+    /// The imports are sorted by UTF-32 codepoints without case folding
+    UString sortingKey(Import *import) {
+        return import->file->value;
+    }
+
+    /// Check if `local` expression is used for importing,
+    bool isGoodLocal(Local *local) {
+        for (const auto &bind: local->binds) {
+            if (bind.body->type != AST_IMPORT) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    Local *goodLocalOrNull(AST *expr) {
+        if (auto *local = dynamic_cast<Local*>(expr)) {
+            return isGoodLocal(local) ? local : nullptr;
+        } else {
+            return nullptr;
+        }
+    }
+
+    /** Split openFodder into what logically belongs to the previous AST node
+     * what belongs to the current AST node.
+     *
+     * Example:
+     * local x = import "xxx.jsonnet"; // Importing stuff here
+     *
+     * // blah blah
+     * {}
+     *
+     * In such case "// Importing  stuff here\n" part of the fodder belongs
+     * to the `local` and the rest to the {}.
+     */
+    std::pair<Fodder, Fodder> splitOpenFodder(AST *expr) {
+        const Fodder &fodder = open_fodder(expr);
+        Fodder afterPrev, beforeThis;
+        bool inSecondPart = false;
+        for (const auto &fodderElem: fodder) {
+            if (inSecondPart) {
+                fodder_push_back(beforeThis, fodderElem);
+            } else {
+                afterPrev.push_back(fodderElem);
+            }
+            if (fodderElem.kind != FodderElement::Kind::INTERSTITIAL) {
+                inSecondPart = true;
+                if (fodderElem.blanks > 0) {
+                    afterPrev.back().blanks = 0;
+                    assert(beforeThis.empty());
+                    beforeThis.emplace_back(
+                        FodderElement::Kind::LINE_END,
+                        fodderElem.blanks,
+                        fodderElem.indent,
+                        std::vector<std::string>()
+                    );
+                }
+            }
+        }
+        return {afterPrev, beforeThis};
+    }
+
+    void sortGroup(ImportElems &imports) {
+        // We don't want to change behaviour in such case:
+        // local foo = "b.jsonnet";
+        // local foo = "a.jsonnet";
+        // So we don't change the order when there are shadowed variables.
+        if (!duplicatedVariables(imports)) {
+            std::sort(imports.begin(), imports.end());
+        }
+    }
+
+    ImportElems extractImportElems(Local *local, Fodder after) {
+        ImportElems result;
+        for (auto &bind: local->binds) {
+            Import *import = dynamic_cast<Import*>(bind.body);
+            assert(import != nullptr);
+            result.emplace_back(sortingKey(import), after, &bind);
+            after = {FodderElement(FodderElement::Kind::LINE_END, 0, 0, {})};
+        }
+        return result;
+    }
 
+    AST *buildGroupAST(ImportElems &imports, AST *body, const Fodder &groupOpenFodder) {
```

Comment:
Well, existing code often goes over 80, so I didn't worry about it very much. @sparkprime?